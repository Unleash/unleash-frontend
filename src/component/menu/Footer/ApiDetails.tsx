import { ReactElement } from 'react';
import ConditionallyRender from '../../common/ConditionallyRender';
import { IVersionInfo } from '../../../interfaces/uiConfig';
import { formatCurrentVersion, formatUpdateNotification } from './ApiDetails.helpers';

export interface IPartialUiConfig {
	name: string;
	version: string;
	slogan?: string;
	environment?: string;
	versionInfo?: IVersionInfo;
}

interface IApiDetailsProps {
	uiConfig: IPartialUiConfig;
}

export const ApiDetails = (props: IApiDetailsProps): ReactElement => {
	const instanceId = props.uiConfig.versionInfo?.instanceId;
	const currentVersion = formatCurrentVersion(props.uiConfig);
	const environment = props.uiConfig.environment;
	const updateNotification = formatUpdateNotification(props.uiConfig);
	
	return (
		<section title="API details">
			<h4>
				{currentVersion}{' '}
				<ConditionallyRender
					condition={Boolean(environment)}
					show={<small>({environment})</small>}
				/>
			</h4>
			<ConditionallyRender
				condition={Boolean(updateNotification)}
				show={
					<small>
						{updateNotification}
						<br />
					</small>
				}
			/>
			<br />
			<small>{props.uiConfig.slogan}</small>
			<br />
			<ConditionallyRender
				condition={Boolean(instanceId)}
				show={<small>{`${instanceId}`}</small>}
			/>
		</section>
	);
};
