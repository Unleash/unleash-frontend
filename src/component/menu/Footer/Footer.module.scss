.list {
    padding: 0;
    margin: 0;
}

.listItem {
    padding: 0;
    margin: 0;
}

.listItem a {
    text-decoration: none;
    color: #fff;
}