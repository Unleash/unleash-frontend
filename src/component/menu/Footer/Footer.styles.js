import { makeStyles } from '@material-ui/styles';

export const useStyles = makeStyles(theme => ({
    footer: {
        background: theme.palette.footer.background,
        padding: '2rem 4rem',
        width: '100%',
        flexGrow: 1,
        zIndex: 100,
    },
    list: {
        padding: 0,
        margin: 0,
    },
    listItem: {
        padding: 0,
        margin: 0,
        '& a': {
            textDecoration: 'none',
            color: theme.palette.footer.main,
        },
    },
}));
