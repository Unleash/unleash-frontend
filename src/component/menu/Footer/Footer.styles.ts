import makeStyles from '@mui/styles/makeStyles';

export const useStyles = makeStyles(theme => ({
    footer: {
        // @ts-expect-error
        background: theme.palette.footer.background,
        padding: '2rem 4rem',
        width: '100%',
        flexGrow: 1,
        zIndex: 100,
    },
    list: {
        padding: 0,
        margin: 0,
    },
    listItem: {
        padding: 0,
        margin: 0,
        '& a': {
            textDecoration: 'none',
            // @ts-expect-error
            color: theme.palette.footer.main,
        },
    },
}));
