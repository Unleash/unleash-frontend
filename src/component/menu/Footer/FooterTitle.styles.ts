import { makeStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles(theme => ({
    title: {
        all: 'unset',
        display: 'block',
        margin: '1rem 0',
        fontSize: '1rem',
        fontWeight: theme.fontWeight.bold,
    },
}));
