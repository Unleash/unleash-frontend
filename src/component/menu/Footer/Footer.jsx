/* eslint-disable react/jsx-no-target-blank */

import { List, ListItem, ListItemText, Grid } from '@material-ui/core';

import ShowApiDetailsContainer from '../../api/show-api-details-container';

import { useStyles } from './Footer.styles';

export const Footer = () => {
    const styles = useStyles();

    return (
        <footer className={styles.footer}>
            <Grid container justifyContent="center" spacing={10}>
                <Grid item md={4} xs={12}>
                    <ShowApiDetailsContainer />
                </Grid>
                <Grid item xs={12} md="auto">
                    <Grid container spacing={7} direction="row">
                        <Grid item>
                            <section title="Unleash SDK">
                                <h4>Server SDKs</h4>
                                <List className={styles.list} dense>
                                    <ListItem className={styles.listItem}>
                                        <ListItemText
                                            primary={
                                                <a
                                                    href="https://docs.getunleash.io/sdks/node_sdk"
                                                    className={styles.link}
                                                    target="_blank"
                                                >
                                                    Node.js
                                                </a>
                                            }
                                        />
                                    </ListItem>
                                    <ListItem className={styles.listItem}>
                                        <ListItemText
                                            primary={
                                                <a
                                                    href="https://docs.getunleash.io/sdks/java_sdk"
                                                    className={styles.link}
                                                    target="_blank"
                                                >
                                                    Java
                                                </a>
                                            }
                                        />
                                    </ListItem>
                                    <ListItem className={styles.listItem}>
                                        <ListItemText
                                            primary={
                                                <a
                                                    href="https://docs.getunleash.io/sdks/go_sdk"
                                                    className={styles.link}
                                                    target="_blank"
                                                >
                                                    Go
                                                </a>
                                            }
                                        />
                                    </ListItem>{' '}
                                    <ListItem className={styles.listItem}>
                                        <ListItemText
                                            primary={
                                                <a
                                                    href="https://docs.getunleash.io/sdks/ruby_sdk"
                                                    className={styles.link}
                                                    target="_blank"
                                                >
                                                    Ruby
                                                </a>
                                            }
                                        />
                                    </ListItem>{' '}
                                    <ListItem className={styles.listItem}>
                                        <ListItemText
                                            primary={
                                                <a
                                                    href="https://docs.getunleash.io/sdks/python_sdk"
                                                    className={styles.link}
                                                    target="_blank"
                                                >
                                                    Python
                                                </a>
                                            }
                                        />
                                    </ListItem>
                                    <ListItem className={styles.listItem}>
                                        <ListItemText
                                            primary={
                                                <a
                                                    href="https://docs.getunleash.io/sdks/dot_net_sdk"
                                                    className={styles.link}
                                                    target="_blank"
                                                >
                                                    .NET
                                                </a>
                                            }
                                        />
                                    </ListItem>
                                    <ListItem className={styles.listItem}>
                                        <ListItemText
                                            primary={
                                                <a
                                                    href="https://docs.getunleash.io/sdks/php_sdk"
                                                    className={styles.link}
                                                    target="_blank"
                                                >
                                                    PHP
                                                </a>
                                            }
                                        />
                                    </ListItem>
                                    <ListItem className={styles.listItem}>
                                        <ListItemText
                                            primary={
                                                <a
                                                    href="https://docs.getunleash.io/sdks"
                                                    className={styles.link}
                                                    target="_blank"
                                                >
                                                    All SDKs
                                                </a>
                                            }
                                        />
                                    </ListItem>
                                </List>
                            </section>
                        </Grid>
                        <Grid item>
                            <section title="Unleash SDK">
                                <h4>Frontend SDKs</h4>
                                <List className={styles.list} dense>
                                    <ListItem className={styles.listItem}>
                                        <ListItemText
                                            primary={
                                                <a
                                                    href="https://docs.getunleash.io/sdks/unleash-proxy"
                                                    className={styles.link}
                                                    target="_blank"
                                                >
                                                    Unleash Proxy
                                                </a>
                                            }
                                        />
                                    </ListItem>
                                    <ListItem className={styles.listItem}>
                                        <ListItemText
                                            primary={
                                                <a
                                                    href="https://docs.getunleash.io/sdks/proxy-javascript"
                                                    className={styles.link}
                                                    target="_blank"
                                                >
                                                    JavaScript SDK
                                                </a>
                                            }
                                        />
                                    </ListItem>
                                    <ListItem className={styles.listItem}>
                                        <ListItemText
                                            primary={
                                                <a
                                                    href="https://docs.getunleash.io/sdks/proxy-react"
                                                    className={styles.link}
                                                    target="_blank"
                                                >
                                                    React SDK
                                                </a>
                                            }
                                        />
                                    </ListItem>
                                    <ListItem className={styles.listItem}>
                                        <ListItemText
                                            primary={
                                                <a
                                                    href="https://docs.getunleash.io/sdks/proxy-ios"
                                                    className={styles.link}
                                                    target="_blank"
                                                >
                                                    iOS SDK
                                                </a>
                                            }
                                        />
                                    </ListItem>
                                    <ListItem className={styles.listItem}>
                                        <ListItemText
                                            primary={
                                                <a
                                                    href="https://docs.getunleash.io/sdks/android_proxy_sdk"
                                                    className={styles.link}
                                                    target="_blank"
                                                >
                                                    Android SDK
                                                </a>
                                            }
                                        />
                                    </ListItem>
                                </List>
                            </section>
                        </Grid>
                        <Grid item>
                            <section>
                                <h4>About</h4>
                                <List className={styles.list} dense>
                                    <ListItem className={styles.listItem}>
                                        <ListItemText
                                            primary={
                                                <a
                                                    href="https://www.getunleash.io/"
                                                    className={styles.link}
                                                    target="_blank"
                                                >
                                                    getunleash.io
                                                </a>
                                            }
                                        />
                                    </ListItem>
                                    <ListItem className={styles.listItem}>
                                        <ListItemText
                                            primary={
                                                <a
                                                    href="https://twitter.com/getunleash"
                                                    className={styles.link}
                                                    target="_blank"
                                                >
                                                    Twitter
                                                </a>
                                            }
                                        />
                                    </ListItem>
                                    <ListItem className={styles.listItem}>
                                        <ListItemText
                                            primary={
                                                <a
                                                    href="https://www.linkedin.com/company/getunleash"
                                                    className={styles.link}
                                                    target="_blank"
                                                >
                                                    LinkedIn
                                                </a>
                                            }
                                        />
                                    </ListItem>
                                    <ListItem className={styles.listItem}>
                                        <ListItemText
                                            primary={
                                                <a
                                                    href="https://github.com/Unleash/unleash"
                                                    className={styles.link}
                                                    target="_blank"
                                                >
                                                    GitHub
                                                </a>
                                            }
                                        />
                                    </ListItem>
                                    <ListItem className={styles.listItem}>
                                        <ListItemText
                                            primary={
                                                <a
                                                    href="https://slack.unleash.run"
                                                    className={styles.link}
                                                    target="_blank"
                                                >
                                                    Slack Community
                                                </a>
                                            }
                                        />
                                    </ListItem>
                                </List>
                            </section>
                        </Grid>
                    </Grid>
                </Grid>
            </Grid>
        </footer>
    );
};

export default Footer;
