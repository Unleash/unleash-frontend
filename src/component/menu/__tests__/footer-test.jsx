import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import Footer from "../Footer/Footer";

jest.mock("@material-ui/core");

test("should render DrawerMenu", () => {
    const tree = renderer.create(
        <MemoryRouter>
            <Footer />
        </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
});

test('should render DrawerMenu with "features" selected', () => {
    const tree = renderer.create(
        <MemoryRouter initialEntries={["/features"]}>
            <Footer />
        </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
});
