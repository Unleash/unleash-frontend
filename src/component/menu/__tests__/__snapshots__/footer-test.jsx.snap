// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DrawerMenu 1`] = `
<react-mdl-FooterSection
  type="middle"
>
  <react-mdl-FooterDropDownSection
    title="Menu"
  >
    <react-mdl-FooterLinkList>
      <a
        aria-current={null}
        href="/features"
        onClick={[Function]}
      >
        Feature Toggles
      </a>
      <a
        aria-current={null}
        href="/strategies"
        onClick={[Function]}
      >
        Strategies
      </a>
      <a
        aria-current={null}
        href="/history"
        onClick={[Function]}
      >
        Event History
      </a>
      <a
        aria-current={null}
        href="/archive"
        onClick={[Function]}
      >
        Archived Toggles
      </a>
      <a
        aria-current={null}
        href="/applications"
        onClick={[Function]}
      >
        Applications
      </a>
      <a
        aria-current={null}
        href="/tag-types"
        onClick={[Function]}
      >
        Tag types
      </a>
      <a
        aria-current={null}
        href="/logout"
        onClick={[Function]}
      >
        Sign out
      </a>
      <a
        href="https://github.com/Unleash/unleash/"
        target="_blank"
      >
        GitHub
      </a>
    </react-mdl-FooterLinkList>
  </react-mdl-FooterDropDownSection>
  <react-mdl-FooterDropDownSection
    title="Client SDKs"
  >
    <react-mdl-FooterLinkList>
      <a
        href="https://github.com/Unleash/unleash-client-node/"
      >
        Node.js
      </a>
      <a
        href="https://github.com/Unleash/unleash-client-java/"
      >
        Java
      </a>
      <a
        href="https://github.com/Unleash/unleash-client-go/"
      >
        Go
      </a>
      <a
        href="https://github.com/unleash/unleash-client-ruby"
      >
        Ruby
      </a>
      <a
        href="https://github.com/Unleash/unleash-client-python"
      >
        Python
      </a>
      <a
        href="https://github.com/Unleash/unleash-client-core"
      >
        .Net Core
      </a>
      <a
        href="https://unleash.github.io/docs/client_sdk"
      >
        All client SDKs
      </a>
    </react-mdl-FooterLinkList>
  </react-mdl-FooterDropDownSection>
</react-mdl-FooterSection>
`;

exports[`should render DrawerMenu with "features" selected 1`] = `
<react-mdl-FooterSection
  type="middle"
>
  <react-mdl-FooterDropDownSection
    title="Menu"
  >
    <react-mdl-FooterLinkList>
      <a
        aria-current="page"
        className="active"
        href="/features"
        onClick={[Function]}
        style={Object {}}
      >
        Feature Toggles
      </a>
      <a
        aria-current={null}
        href="/strategies"
        onClick={[Function]}
      >
        Strategies
      </a>
      <a
        aria-current={null}
        href="/history"
        onClick={[Function]}
      >
        Event History
      </a>
      <a
        aria-current={null}
        href="/archive"
        onClick={[Function]}
      >
        Archived Toggles
      </a>
      <a
        aria-current={null}
        href="/applications"
        onClick={[Function]}
      >
        Applications
      </a>
      <a
        aria-current={null}
        href="/tag-types"
        onClick={[Function]}
      >
        Tag types
      </a>
      <a
        aria-current={null}
        href="/logout"
        onClick={[Function]}
      >
        Sign out
      </a>
      <a
        href="https://github.com/Unleash/unleash/"
        target="_blank"
      >
        GitHub
      </a>
    </react-mdl-FooterLinkList>
  </react-mdl-FooterDropDownSection>
  <react-mdl-FooterDropDownSection
    title="Client SDKs"
  >
    <react-mdl-FooterLinkList>
      <a
        href="https://github.com/Unleash/unleash-client-node/"
      >
        Node.js
      </a>
      <a
        href="https://github.com/Unleash/unleash-client-java/"
      >
        Java
      </a>
      <a
        href="https://github.com/Unleash/unleash-client-go/"
      >
        Go
      </a>
      <a
        href="https://github.com/unleash/unleash-client-ruby"
      >
        Ruby
      </a>
      <a
        href="https://github.com/Unleash/unleash-client-python"
      >
        Python
      </a>
      <a
        href="https://github.com/Unleash/unleash-client-core"
      >
        .Net Core
      </a>
      <a
        href="https://unleash.github.io/docs/client_sdk"
      >
        All client SDKs
      </a>
    </react-mdl-FooterLinkList>
  </react-mdl-FooterDropDownSection>
</react-mdl-FooterSection>
`;
